### DISCUSSION TITLE
Refer to a script in a parent directory

### DISCUSSION THREAD

**Discussion by erossini** [pester/Pester#2419] [Q&A] [answered]
To organize better the scripts and the relative tests, I created in my _pipeline_ a simple structure: the _scripts_ folder has all the scripts; then the _tests_ folder contains the folder for each script and the necessary files.

![image](https://github.com/pester/Pester/assets/9497415/fec7f978-dbbe-4cc6-82fb-dfdc6029598c)

I have an issue in find the script from the test in the setup. For example, the code is

```powershell
Describe "Setup" {
    BeforeAll {
        . $PSCommandPath.Replace('.Tests.ps1', '.ps1') -flowJsonPath $flowJsonPath
    }
}
```

Also, I tried to write the setup like this

```powershell
Describe "Setup" {
    BeforeAll {
        $scriptName = [System.IO.Path]::GetFileName($PSCommandPath.Replace('.Tests.ps1', '.ps1'))
        $scriptPath = Split-Path -Parent $PSCommandPath.Replace('.Tests.ps1', '.ps1')
        $scriptPath = $scriptPath -replace "Tests\\JsonFiltering", ""
        $script = [System.IO.Path]::Combine($scriptPath, $scriptName)
        . $script -flowJsonPath $flowJsonPath -flowTycheJsonPath $flowTycheJsonPath -outputFilePath $outputFilePath
    }
}
```

but I still get an error

> CommandNotFoundException: The term 'C:\Users\A084\Documents\Projects\MyProject\pipelines\Scripts\Tests\JsonFiltering\JsonFilte
ring.ps1' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.

Is there a way to use a script in the parent directory?

**Comment by johlju** [comment, marked as answer]
The variable `$PSScriptRoot`. So `. $PSScriptRoot/../JsonFiltering.ps1`.

**Comment by erossini** [comment]
Thank you for answer. If it can be useful, here the code 

```powershell
$scriptRoot = "$PSScriptRoot/../../"
$filename = (Split-Path $PSCommandPath -Leaf).Replace('.Tests.ps1', '.ps1')
$scriptToTest = [System.IO.Path]::Combine($scriptRoot, $filename)
```

**Reply by fflaten** [reply]
Hi. Does this mean your issue was resolved? If so, please mark as answer 🙂

**Comment by billgothacked** [comment]
What's up: )
